python/plugins/processing/algs/qgis/Heatmap.py

```diff
@@ -170,7 +204,7 @@ def processAlgorithm(self, parameters, context, feedback):
         kde = QgsKernelDensityEstimation(kde_params, output, output_format)
 
         if kde.prepare() != QgsKernelDensityEstimation.Success:
-            raise GeoAlgorithmExecutionException(
+            raise QgsProcessingException(
```

Comment:
it should be fine to keep throwing GeoAlgorithmExecutionException - there's code which handles this and maps it automatically to the c++ exception. I guess there's no harm either way